src/app/components/layouts/card/terra-card.component.spec.ts

```diff
@@ -1,3 +1,199 @@
+        fixture.detectChanges();
+        debugElement = fixture.debugElement;
+    });
+
+    afterEach(() =>
+    {
+        cardComponent.inputPlaceholderIcon = null;
+        cardComponent.inputImagePath = null;
+    });
+
+    it('should create', () =>
+    {
+        expect(component).toBeTruthy();
+    });
+
+    it('should initialize its inputs', () =>
+    {
+        expect(cardComponent.inputIsSelected).toBe(false);
+        expect(cardComponent.inputImagePath).toBeUndefined();
+        expect(cardComponent.inputPlaceholderIcon).toBeUndefined();
+    });
+
+    describe('footer content', () =>
+    {
+        let footerElement:DebugElement;
+
+        beforeEach(() =>
+            {
+                footerElement = debugElement.query(By.css('div.card-footer'));
+            }
+        );
+
+        it('should div-element for footer be shown if content is given', () =>
+        {
+            let ngContentElement:DebugElement = footerElement.query(By.css('p'));
+            let footerContentElement:HTMLElement = ngContentElement.nativeElement;
+            expect(footerElement).toBeTruthy();
+            expect(cardComponent.viewChildFooter).toBeTruthy();
+            expect(cardComponent.viewChildFooter.nativeElement.children.length).toBeGreaterThan(0);
+            expect(footerContentElement.innerHTML).toEqual('card footer');
+            expect(footerElement.nativeElement.hidden).toBe(false);
+        });
+
+        it('should set class selected depending on #inputIsSelected', () =>
+        {
+            let terraCardElement:DebugElement = debugElement.query(By.css('div.terra-card'));
+            expect(footerElement.classes['selected']).toBeFalsy();
+            expect(terraCardElement.classes['selected']).toBeFalsy();
+
+            cardComponent.inputIsSelected = true;
+            fixture.detectChanges();
+            expect(footerElement.classes['selected']).toBeTruthy();
+            expect(terraCardElement.classes['selected']).toBeTruthy();
+        });
+    });
+
+    describe('header content', () =>
+    {
+        it('should show div-element for header if content is given', () =>
+        {
+            let headerElement:DebugElement = debugElement.query(By.css('div.card-header'));
+            let ngContentElement:DebugElement = headerElement.query(By.css('p'));
+            let headerContentElement:HTMLElement = ngContentElement.nativeElement;
+            expect(headerElement).toBeTruthy();
+            expect(cardComponent.viewChildHeader).toBeTruthy();
+            expect(cardComponent.viewChildHeader.nativeElement.children.length).toBeGreaterThan(0);
+            expect(headerContentElement.innerHTML).toEqual('card header');
+            expect(headerElement.nativeElement.hidden).toBe(false);
+        });
+    });
+
+    describe('block content', () =>
+    {
+        it('should show div-element for content if content is given', () =>
```

Comment:
TerraCardComponent block content should show div-element for content if content is given

hört sich verwirrend an